verification/src/solidity/compiler_fetcher.rs

```diff
@@ -104,15 +116,75 @@ impl TryFrom<(json::CompilerInfo, &Url)> for CompilerInfo {
     }
 }
 
+impl CompilerVersions {
+    fn spawn_refresh_job(self, versions_list_url: Url, cron_schedule: Schedule) {
+        log::info!("spawn version refresh job");
+        spawn_job_with_schedule(cron_schedule, move || {
+            let versions_list_url = versions_list_url.clone();
+            let versions = self.clone();
+            async move {
+                let refresh_result = versions.refresh_versions(&versions_list_url).await;
+                if let Err(err) = refresh_result {
+                    log::error!("error during version refresh: {}", err);
+                };
+            }
+        });
+    }
+
+    async fn refresh_versions(&self, versions_list_url: &Url) -> anyhow::Result<()> {
+        log::info!("looking for new compilers versions");
+        let fetched_versions = try_fetch_versions(versions_list_url)
+            .await
+            .map_err(anyhow::Error::msg)?;
+        let need_to_update = {
+            let versions = self.0.read();
+            fetched_versions != *versions
+        };
+        if need_to_update {
+            let (old_len, new_len) = {
+                let mut versions = self.0.write();
```

Comment:
where is mah comment
something something `// we don't need to check condition again, we can just override the value`